app/src/main/java/com/xmartlabs/template/controller/ServiceAndDatabaseController.java

```diff
@@ -0,0 +1,140 @@
+package com.xmartlabs.template.controller;
+
+import android.support.annotation.CheckResult;
+import android.support.annotation.NonNull;
+
+import com.raizlabs.android.dbflow.sql.language.property.Property;
+import com.xmartlabs.template.helper.function.BiFunction;
+import com.xmartlabs.template.helper.function.Function;
+import com.xmartlabs.template.model.EntityWithId;
+
+import java.util.List;
+
+import io.reactivex.Completable;
+import io.reactivex.Flowable;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+@SuppressWarnings({"unused", "WeakerAccess"})
+public abstract class ServiceAndDatabaseController<Id, D extends EntityWithId<Id>, C> extends Controller {
+  private final DatabaseProvider<Id, D, C> databaseProvider;
+  private final ServiceProvider<Id, D> serviceProvider;
+
+  @CheckResult
+  @NonNull
+  @SuppressWarnings("unchecked")
+  protected Flowable<List<D>> getEntities(@NonNull Single<List<D>> serviceCall, @NonNull C... conditions) {
+    //noinspection unchecked
+    return Flowable
+        .concatArrayDelayError(
+            databaseProvider.getEntities(conditions).toFlowable(),
+            serviceProvider.getEntities(serviceCall).toFlowable()
+        )
+        .subscribeOn(Schedulers.io())
+        .observeOn(Schedulers.io())
+        .scan((databaseEntities, serviceEntities) ->
+            databaseProvider.updateEntities(serviceEntities, conditions).blockingGet()
+        )
+        .observeOn(AndroidSchedulers.mainThread());
+  }
+
+  @CheckResult
+  @NonNull
+  protected Observable<D> getEntity(@NonNull Function<Id, Single<D>> serviceCall, Id id) {
```

Comment:
What do you think about implementing some kind of cache policy? For example, I want to fetch an entity just from DB, because I'm sure that it was recently updated and want to prevent an extra service call?